src/core/execute.c

```diff
@@ -872,8 +872,10 @@ static int setup_pam(
 
         STRV_FOREACH(e, *env) {
                 pam_code = pam_putenv(handle, *e);
-                if (pam_code != PAM_SUCCESS)
+                if (pam_code != PAM_SUCCESS) {
+                        e = NULL;
```

Comment:
A similar pattern occurs slightly [above](https://github.com/systemd/systemd/blob/master/src/core/execute.c#L863) this code, but if you rather see a `char **nv` added, that is okay too. I'll update the PR later today.